src/main/java/com/cgdpd/library/dto/BookCopy/BookCopyDTO.java

```diff
@@ -0,0 +1,38 @@
+package com.cgdpd.library.dto.BookCopy;
+
+import static com.cgdpd.library.validation.Validator.checkTrackingStatus;
+import static com.cgdpd.library.validation.Validator.required;
+
+import com.cgdpd.library.entity.TrackingStatus;
+import java.util.Optional;
+import lombok.Builder;
+
+@Builder
+public record BookCopyDTO(Long id,
+                          Long bookId,
+                          TrackingStatus trackingStatus,
+                          Optional<Long> userId) {
+
+    public BookCopyDTO(Long id,
+                       Long bookId,
+                       TrackingStatus trackingStatus,
+                       Optional<Long> userId) {
+        this.id = required("id", id);
+        this.bookId = required("bookId", bookId);
+        this.trackingStatus = required("trackingStatus", trackingStatus);
+        this.userId = checkTrackingStatus("userId", trackingStatus, userId);
+    }
+
+    public static BookCopyDTO create(Long id, Long bookId) {
+        return new BookCopyDTO(id, bookId, TrackingStatus.AVAILABLE, Optional.empty());
+    }
+
+    public BookCopyDTO withStatus(TrackingStatus newStatus) {
+        return new BookCopyDTO(this.id, this.bookId, newStatus, this.userId);
+    }
```

Comment:
This is not following the rich domain model https://www.baeldung.com/java-anemic-vs-rich-domain-objects.
It is basically leaving up to whatever is calling this methods to decide the status and not checking if it is allowed.

Create methods to change the status and make sure that the status change is possible, create validation method here to check that, also don't use the constructor directly use the builder as it is more readable 

Create this method to create a new copy of the object with the same state
```
    private BookCopyDTO.BookCopyDTOBuilder copy() {
        return builder()
              .id(id)
              .bookId(bookId)
              .trackingStatus(trackingStatus)
              .userId(userId);
    }
```

Also initial state is not `AVAILABLE` it should be `BEING_PROCESSED`